backend/src/util/VerifyFIN.ts

```diff
@@ -0,0 +1,34 @@
+const checkAlpha = ['X', 'W', 'U', 'T', 'R', 'Q', 'P', 'N', 'M', 'L', 'K'];
+const weightArray = [2, 7, 6, 5, 4, 3, 2];
+
+function validateFIN(fin: string): string {
+  const finUpper = fin.trim.toUpperCase();
+  if (finUpper.length === 9) {
+    const sumArray = fin.slice(1, 8).split('').map(Number); // FIN String into array of numbers to multiply by weight
+
+    let sum = 0;
+    for (let i = 0; i < sumArray.length; i += 1) {
+      sum += sumArray[i] * weightArray[i];
+    } // Cross multiply with weightArray to get sum
+
+    if (finUpper[0] === 'G') {
+      sum += 4;
+    } // FIN that starts with G has their checkAlpha shifted by 4
+
+    if (checkAlpha[sum % 11] === finUpper.slice(-1)) {
```

Comment:
Collapse this if..return into a single return statement e.g. return actual === expected

Reply:
What is this magic number 11? If it is the length of the checkSum array, use that value instead

Reply:
The magic number is part of the algorithm to mod the sum by 11. Not sure why they use 11 too..

Reply:
Hmm.... Ok I'm comfortable keeping this as just `11`. I'm fairly certain it's the length of the array (because that's a common way of computing checksums), but it's better not to make assumptions.